### PR TITLE
[#138683563] Adds booking channel tag

### PR DESCRIPTION
**Pull request by pcsouza**
Creates a booking channel tag. This tag will contain the 3 forms of
identifying the reservation with the booking channel as well as the name
of the booking channel.

Created an "IsPermanent" field on reservation tags. This field denotes
tags that can be established in genesis, but cannot be removed by
amendment. For the forseeable future, I don't see a case for supporting
amending channel information on a reservation. Having this "IsPermament"
tag will allow the HOLMS system from Model to Client to enforce that
permanent tags are permanent.

Added channel identification information onto mail reservation DTOs. The
plan is to store the two primary booking ids (channel manager, and source
channel ids) as top-level fields on the reservation. This is for queries
where we need to look up reservations by one of these IDs.

Added the OTASyncSvc app service. This will allow us to hook the
OTASyncSvc up to the scheduler, or some other runner if we so desire.

### CODE REVIEW

File: csharp/HOLMS.Platform/HOLMS.Platform/Support/ReservationTags/CompStayTag.cs [hunk 1]
```diff
@@ -1,9 +1,17 @@
-﻿namespace HOLMS.Platform.Support.ReservationTags {
+﻿using System;
+
+namespace HOLMS.Platform.Support.ReservationTags {
     public class CompStayTag : ReservationTagBase {
         public CompStayTag() { }
 
         public CompStayTag(string[] tokens) { }
 
+        public override bool IsPermanent {
```
Comment (eldavido) [inline csharp/HOLMS.Platform/HOLMS.Platform/Support/ReservationTags/CompStayTag.cs]: This makes me a little leery..."permanence" is more of a booking/amendment concern, I'm not sure it's appropriate to expose this to other bits of the system.

I don't absolutely hate it, but let's be careful not to let too much policy "leak out" from the booking context. The intention is that the various contexts can use the tags are pure data, without worrying about sharing too much behavior.

File: csharp/HOLMS.Platform/HOLMS.Platform/Support/ReservationTags/ReservationTagBase.cs [hunk 2]
```diff
@@ -19,6 +20,8 @@ public abstract class ReservationTagBase {
                 return string.Join(":", new List<string> { CompStayCategory }.Concat(dp));
             } else if (this is GroupBookingTag) {
                 return string.Join(":", new List<string> { GroupBookingCategory }.Concat(dp));
+            } else if (this is OTABookingTag) {
```
Comment (eldavido) [inline csharp/HOLMS.Platform/HOLMS.Platform/Support/ReservationTags/ReservationTagBase.cs]: This is textbook "switched polymorphism", can you please move this to the subclasses?

If you run into trouble with `ToString()`, just override it and have it call a different method.

**Comment by pcsouza** [comment]
Per slack discussion, leaving IsPermanent on the tag. It's useful for driving behavior of the lodgeic client in amending reservations.

**Comment by pcsouza** [comment]
@eldavido Made changes as requested. PTAL

**Pull request merged**